docs/tutorials/basic/web.md

```diff
@@ -0,0 +1,219 @@
+layout: docs
+headline: gRPC Basics - Web
+sidenav: doc-side-tutorial-nav.html
+type: markdown
+---
+<p class="lead">This tutorial provides a basic introduction on how to use
+gRPC-Web from browsers.</p>
+
+By walking through this example you'll learn how to:
+
+- Define a service in a .proto file.
+- Generate client code using the protocol buffer compiler.
+- Use the gRPC-Web API to write a simple client for your service.
+
+It assumes a passing familiarity with [protocol
+buffers](https://developers.google.com/protocol-buffers/docs/overview).
+
+<div id="toc"></div>
+
+<a name="why-grpc"></a>
+
+## Why use gRPC?
+
+With gRPC you can define your service once in a .proto file and implement
+clients and servers in any of gRPC's supported languages, which in turn can be
+run in environments ranging from servers inside Google to your own tablet - all
+the complexity of communication between different languages and environments is
+handled for you by gRPC. You also get all the advantages of working with
+protocol buffers, including efficient serialization, a simple IDL, and easy
+interface updating.
+
+
+<a name="setup"></a>
+
+## Define the Service
+
+The first step when creating a gRPC service is to define the service methods
+and their request and response message types using protocol buffers. In this
+example, we define our `EchoService` in a file called
+[`echo.proto`](https://github.com/grpc/grpc-web/blob/master/net/grpc/gateway/examples/echo/echo.proto).
+For more information about protocol buffers and proto3 syntax, please see the
+[protobuf documentation][].
+
+```protobuf
+message EchoRequest {
+  string message = 1;
+}
+
+message EchoResponse {
+  string message = 1;
+}
+
+service EchoService {
+  rpc Echo(EchoRequest) returns (EchoResponse);
+}
+```
+
+## Implement gRPC Backend Server
+
+Next, we implement our EchoService interface using C++ in the backend gRPC
+`EchoServer`. This will handle requests from clients. See the file
+[`echo_server.cc`](https://github.com/grpc/grpc-web/blob/master/net/grpc/gateway/examples/echo/echo_server.cc)
+for details.
+
+You can implement the server in any language supported by gRPC. Please see
+the [main page][] for more details.
+
+```cpp
+Status EchoServiceImpl::Echo(ServerContext* context,
+                             const EchoRequest* request,
+                             EchoResponse* response) {
+  response->set_message(request->message());
+  return Status::OK;
+}
+```
+
+
+## Configure the Envoy Proxy
+
+In this example, we will use the [Envoy](https://www.envoyproxy.io/)
+proxy to forward the gRPC browser request to the backend server. You can see
+the complete config file in
+[envoy.yaml](https://github.com/grpc/grpc-web/blob/master/net/grpc/gateway/examples/echo/envoy.yaml)
+
+To forward the gRPC requests to the backend server, we need a block like
+this:
+
+```yaml
+  listeners:
+  - name: listener_0
+    address:
+      socket_address: { address: 0.0.0.0, port_value: 8080 }
```

Comment:
In the future, I think there is a DNS plugin for Envoy that let's you bind to localhost.

Reply:
Will keep that in mind, thanks!